### PR TITLE
Bump pipeline to 0.24.1 and knative pkg to 0.22

### PR DESCRIPTION
**Pull request by savitaashture**
# Changes

/cc @dibyom 

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#tests) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#docs) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commits)
- [ ] Release notes block has been filled in or deleted (only if no user facing changes)

_See [the contribution guide](https://github.com/tektoncd/triggers/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

```release-note
Update knative dependency to 0.22 and Pipeline to 0.24.1
```

### CODE REVIEW

File: pkg/apis/triggers/v1alpha1/trigger_template_validation_test.go [hunk 1]
```diff
@@ -115,28 +115,6 @@ func TestTriggerTemplate_Validate(t *testing.T) {
 		template *v1alpha1.TriggerTemplate
 		want     *apis.FieldError
 	}{{
-		name: "invalid objectmetadata, name with dot",
```
Comment (dibyom) [inline pkg/apis/triggers/v1alpha1/trigger_template_validation_test.go:118]: Was this intentional?

Comment (savitaashture) [reply pkg/apis/triggers/v1alpha1/trigger_template_validation_test.go:118]: when i update dependency i see we are not getting below error
```
Invalid resource name: special character . must not be present
```
when there is a `.` in objectmeta name

Comment (dibyom) [reply pkg/apis/triggers/v1alpha1/trigger_template_validation_test.go:118]: hmmm....that's interesting.

Comment (dibyom) [reply pkg/apis/triggers/v1alpha1/trigger_template_validation_test.go:118]: Found it: https://github.com/tektoncd/pipeline/issues/3891

We rely on pipelines for object meta validation. And pipelines recently started allowing dots in object names.

**Pull request merged**